### ISSUE TITLE
Код-ревью

### ISSUE THREAD

**Issue by alexlebedevkz** [aidakamalova-dev/olist-customer-satisfaction#1]
Привет! Посмотрел ноутбук: он хорошо структурирован и легко читается (SQL-витрина → EDA → модели → выводы). Ниже замечания, сгруппированные по важности. Первые пункты влияют на метрики и выводы, поэтому их стоит поправить до того, как опираться на результаты.

### 1. Дубли заказов из-за отзывов (SQL-витрина)
В `olist_order_reviews_dataset` 99 224 отзыва на 98 673 уникальных заказа: у 551 заказа несколько отзывов. `INNER JOIN` размножает строки, а при случайном `train_test_split` копии одного заказа попадают и в train, и в test. Это утечка.

Можно оставить последний отзыв на заказ:

```sql
INNER JOIN (
    SELECT order_id, review_score,
           ROW_NUMBER() OVER (PARTITION BY order_id
                              ORDER BY review_answer_timestamp DESC) AS rn
    FROM olist_order_reviews_dataset
) r ON o.order_id = r.order_id AND r.rn = 1
```

### 2. Разбиение train/test
Данные распределены по времени, и случайный сплит «подглядывает в будущее». Честнее делить по времени: train — заказы до определённой даты по `order_purchase_timestamp`, test — после. Если оставлять случайный сплит, то хотя бы с группировкой по `order_id` (`GroupShuffleSplit`).

### 3. «Основной товар» выбирается случайно
`MIN(product_id) AS main_product_id` берёт первый по алфавиту хэш, а не главный товар заказа. У многотоварных заказов категория и вес получаются фактически случайными. Возможно, отчасти поэтому категория вышла незначимой.

Вариант: брать товар с максимальной ценой (`ROW_NUMBER() OVER (PARTITION BY order_id ORDER BY price DESC)`) или добавить признак «число разных категорий в заказе».

### 4. Метрики не совпадают с выводами EDA
В EDA верно сказано, что при дисбалансе 13/87 accuracy не подходит и нужно смотреть PR-AUC / Recall. Но в коде считается только ROC-AUC, а `classification_report` построен при пороге 0.5. Предлагаю:
- добавить `average_precision_score` (PR-AUC);
- подобрать порог под бизнес-задачу, например precision@k: «сколько недовольных среди топ-5% заказов по риску»;
- сравнить модель с простым правилом «доставили позже плана → недоволен». Без такого бейзлайна непонятно, что даёт ML сверх одного признака.

### 5. Feature importance
Impurity-based `feature_importances_` у Random Forest завышает вес непрерывных признаков с большим числом уникальных значений. Кроме того, важность в модели — это не доля причин недовольства. Формулировку «сроки доставки определяют более 70% причин» лучше заменить на «признаки доставки дают основной вклад в качество модели».

Для более надёжной оценки подойдут `sklearn.inspection.permutation_importance` на тесте или коэффициенты логистической регрессии. Объяснение про `total_items` («путаница при сборке») данными не подтверждено: это гипотеза, так её и стоит подписать.

### 6. Момент прогноза и бизнес-рекомендация
Самые сильные признаки (`delivery_delay_days`, `delivery_time_days`) считаются по фактической дате доставки, поэтому предсказать риск модель может только после доставки. Для проактивной компенсации («до того, как доставка превысила срок») нужен другой набор признаков: плановый срок, расстояние продавец → покупатель, история продавца и т. д. Стоит уточнить формулировку или вынести это в next steps.

## 🟢 Мелочи
- [ ] `!pip install kagglehub` работает только в Colab;
- [ ] импорты повторяются в нескольких ячейках, можно собрать их в одну в начале
- [ ] таблицу geolocation (1 млн строк) можно не загружать, она не используется
- [ ] `total_payment_value` почти равен `total_price + total_freight`, признаки дублируются
- [ ] препроцессинг стоит до заголовка «Блок 3», стоит перенести заголовок выше

Если по какому-то пункту нужны подробности или помощь с кодом, пиши!
